Filter standuper list

BigFinger494 opened the pull request:
Добавил отслеживание любимых стендаперов

BigFinger494 committed:
Добавил возможность выбора стендаперов, за которыми пользователь следит

BigFinger494 committed:
- Добавил удаление стендапера
- Добавил соответствующие сообщения-ответы
- Доработал фильтрование получаемых новостей

yoyrandao commented on src/StandupWatcher/Processing/Notifying/BotFacade.cs:
```diff
@@ -16,138 +16,197 @@
+                    if (isSubscribed)
+                    {
+                        var requestedAuthor = message.Text.Split(' ').ToList();
+                        requestedAuthor = requestedAuthor.Skip(1).ToList();
```
Лучше сплитануть по команде и тримануть первый элемент от получившегося, будет читаемее.

yoyrandao commented on src/StandupWatcher/Startup.cs:
```diff
@@ -46,8 +46,9 @@ public static void ConfigureServices(HostBuilderContext context, IServiceCollect
+			var connStr = _configuration.GetConnectionString("watcherDatabase");
 			services.AddDbContext<DatabaseContext>(
-				options => options.UseNpgsql(_configuration.GetConnectionString("watcherDatabase")));
+				options => options.UseNpgsql(connStr));
```
Это можно вернуть, как было.

yoyrandao commented on src/StandupWatcher/Processing/Notifying/BotFacade.cs:
```diff
@@ -16,138 +16,197 @@
+            _client = client;
+            _logger = logger;
+            _subscribersRepository = subscribersRepository;
+            _subscribedAuthors = subscribedAuthors;
```
_subscribedAuthors**Repository**

yoyrandao commented on src/StandupWatcher/Processing/Notifying/BotFacade.cs:
```diff
@@ -16,138 +16,197 @@
+
+        private SubscribedAuthors newAuthor(long chatId, string standuperName)
+        {
+            var author = new SubscribedAuthors()
+            {
+                CreationTimestamp = DateTime.Now,
+                ModificationTimestamp = DateTime.Now,
+                ChatId = chatId,
+                StanduperName = standuperName
+            };
+
+            return author;
```
1. Можно вернуть сразу new, без лишней переменной
2. Метод наверное стоит переименовать а-ля CreateNewAuthorFilter или что-то типа того
3. Static
4. Смотри, CreationTimestamp и ModificationTimestamp нужны для легкого определения, когда и во сколько что-либо было сделано. Если запись новая, то эти два поля совпадают, потому что запись не была модифицирована. В твоем случае ModificationTimetamp > CreationTimestamp. Когда кто-либо в будущем будет смотреть в бд и диагностировать что-то, он может подумать, что запись поменялась (даже не учитывая, что в нашем проекте нет функционала по апдейту всего этого дела). В общем лучше вынести Datetime.Now в отдельную переменную и ее уже гнать в БД.

yoyrandao commented on src/StandupWatcher/Processing/Notifying/BotFacade.cs:
```diff
@@ -16,138 +16,197 @@
+
+                    SendMessage(message.Chat.Id, isSubscribed ? Messages.SubscribedStatusMessage : Messages.NotSubscribedStatusMessage);
+                }
+                ,
```
А как чувак узнает, на кого у него сейчас фильтр стоит?

yoyrandao commented on src/StandupWatcher/Processing/Notifying/BotFacade.cs:
```diff
@@ -16,138 +16,197 @@
+
+                        if (searchedAuthors.Any())
+                        {
+                            var subscribed = searchedAuthors.First();
```
.FirstOrDefault(), затем проверка на нулл. Нечего лишний раз в бд холостыми запросами плеваться

BigFinger494 replied:
лишнее

yoyrandao commented on src/StandupWatcher/Workers/NotificationWorker.cs:
```diff
@@ -50,7 +53,9 @@ protected override void Process()
 				foreach (var subscriber in subscribers)
 				{
-					foreach (var eventData in payload)
+					var filteredNotifications = payload.Where(regularEvent => subscribedAuthors.Any(authors => regularEvent.Artist.Contains(authors.StanduperName)));
```
Если у человека нет фильтров, то ему придет ничего

yoyrandao reviewed (changes requested):
Я понял, почему конфликт большой такой - у тебя пробелы везде,  а у меня табы. Либо поправь сам, либо вольем так, я сам потом прожимать везде форматтинг буду. Вообще норм, но у меня пара вопросов есть. Например как чел узнает, кто у него уже в фильтрах. А остальное просто импрувы того, что уже есть + пара багов на глаз

yoyrandao commented on src/StandupWatcher/DataAccess/Models/SubscribedAuthors.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace StandupWatcher.DataAccess.Models
+{
+    public record SubscribedAuthors : Entity
```
Если добавил сюда таблицу добавить ее и в скрипт деплоя

BigFinger494 committed:
Вернул подключение, как раньше

BigFinger494 committed:
Дописал "Repository" в переменную

BigFinger494 committed:
Добавил метод "list", доработал обработку команд и новые сообщения

BigFinger494 committed:
сделал табы

BigFinger494 committed:
Merge branch 'main' of https://github.com/yoyrandao/standup-watcher into filterStanduperList

BigFinger494 committed:
Пофиксил метод для создания нового автора

BigFinger494 committed:
Добавил создание таблицы в деплой

BigFinger494 committed:
Добавил новые сообщение в ресы
Исправил оповещение пользователей ни за кем не следящих
Пофиксил обработку .add и .remove
